db/comdb2.c

```diff
@@ -3275,15 +3278,18 @@ static int init_sqlite_table(struct dbenv *dbenv, char *table)
        return -1;
     }
 
+    dyns_init_globals();
     rc = dyns_load_schema_string((char*) schema, dbenv->envname, table);
     if (rc) {
         logmsg(LOGMSG_ERROR, "Can't parse schema for %s\n", table);
-        return -1;
+        rc = -1;
+        goto err;
```

Comment:
I prefer if caller invoked `dyns_init_globals` and invokes `dyns_cleanup_globals` on non-zero return.

Reply:
Well, in this case i tried to keep them as close as possible together that way it appeared as a pattern: dyns_init_globals, dyns_load_schema, dyns_cleanup_globals. 
Anyway, I did reorganize them as much as i could so now most callers will init and cleanup.
There are however places where we do get a lock and this globals init and cleanup needs to be done while the lock is held so I can't clean them all like you wished.